### ISSUE TITLE
Protocol Subclass checking method mentioned in documentation does not work / gives desired output

### ISSUE THREAD

**Issue by david-westreicher** [python/mypy#17799]
**Documentation**

The current documentation [https://mypy.readthedocs.io/en/latest/protocols.html#defining-subprotocols-and-subclassing-protocols](https://mypy.readthedocs.io/en/latest/protocols.html#defining-subprotocols-and-subclassing-protocols) mentions the following:

> explicitly assigning to a protocol instance can be a way to ask the type checker to verify that your class implements a protocol:
> `_proto: SomeProto = cast(ExplicitSubclass, None)`

## Minimal Test
```python
# test.py
from typing import Protocol, cast

class SomeProto(Protocol):
    def method(self) -> str: ...

class ExplicitSubclass(SomeProto):
    pass

_: SomeProto = cast(ExplicitSubclass, None)

```
### Expected Behaviour
```bash
➤ mypy --strict --extra-checks test.py
test.py:12: ERROR  ...
```

### Actual Behaviour
```bash
➤ mypy --version
mypy 1.11.2 (compiled: yes)
➤ mypy --strict --extra-checks test.py
Success: no issues found in 1 source file
```

## Suggestion
Change the documentation example code line to:
```bash
_: type[SomeProto] = ExplicitSubclass
```
This actually triggers a type check of the subclass:
```bash
➤ mypy --strict --extra-checks testmypy.py
testmypy.py:12: error: Can only assign concrete classes to a variable of type "type[SomeProto]"  [type-abstract]
```

**Comment by Sabine22-alt** [comment]
I'll do it !
